lib/Doctrine/ORM/Tools/Pagination/LimitSubqueryOutputWalker.php

```diff
@@ -85,13 +110,27 @@ public function __construct($query, $parserResult, array $queryComponents)
      * Walks down a SelectStatement AST node, wrapping it in a SELECT DISTINCT.
      *
      * @param SelectStatement $AST
+     * @param bool $addMissingItemsFromOrderByToSelect
      *
      * @return string
      *
      * @throws \RuntimeException
      */
-    public function walkSelectStatement(SelectStatement $AST)
+    public function walkSelectStatement(SelectStatement $AST, $addMissingItemsFromOrderByToSelect = true)
```

Comment:
why are you adding an extra argument compared to the walker API ? this looks wrong to me (and it can cause issues later).

The logic needing to be shared between 2 cases should be extracted to a separate method and the case of ``addMissingItemsFromOrderByToSelect`` should use another method than ``walkSelectStatement`` for the case it needs a separate logic